packages/svelte-cloudinary/src/components/CldVideoPlayer.svelte

```diff
@@ -1,214 +1,191 @@
-					...logoOptions,
-					showLogo: true,
-					logoImageUrl: logo.imageUrl,
-					logoOnclickUrl: logo.onClickUrl
-				};
-			}
-
-			// Validation
-			checkCloudinaryCloudName(import.meta.env.VITE_PUBLIC_CLOUDINARY_CLOUD_NAME);
-
-			let playerOptions: CloudinaryVideoPlayerOptions = {
-				autoplayMode: autoPlay,
-				cloud_name: import.meta.env.VITE_PUBLIC_CLOUDINARY_CLOUD_NAME,
-				controls,
-				fontFace: fontFace || '',
-				loop,
-				muted,
-				publicId,
-				secure: true,
-				aspectRatio: `${width}:${height}`,
-				transformation: playerTransformations,
-				...logoOptions
-			};
-
-			if (typeof colors === 'object') {
-				playerOptions.colors = colors;
-			}
-
-			playerRef = cloudinaryRef.videoPlayer(videoRef, playerOptions);
-
-			Object.keys(events).forEach((key) => {
-				if (typeof events[key] === 'function') {
-					playerRef?.on(key, handleEvent);
-				}
-			});
-		}
-	}
-
-	/**
-	 *getPlayerRefs
-	 */
+<script lang="ts">
+	import { getVideoPlayerOptions } from '@cloudinary-util/url-loader';
+	import { loadScript } from '../helpers/scripts';
+	import { mergeGlobalConfig } from '../config';
+	import { onDestroy, onMount } from 'svelte';
 
-	function getPlayerRefs() {
-		return {
-			player: playerRef,
-			video: videoRef
-		};
-	}
+	const PLAYER_VERSION = '1.11.1';
 
-	//@ts-ignore
-	function onLoadError(e) {
-		console.error(`Failed to load Cloudinary Video Player: ${e.message}`);
+	type $$Props = CldVideoPlayerProps;
+	$: ({ config, id, ...videoPlayerOptions } = $$props as CldVideoPlayerProps);
+
+	$: options = getVideoPlayerOptions(
+		videoPlayerOptions,
+		mergeGlobalConfig(config).config,
+	);
+
+	let loaded =
+		typeof window != 'undefined' && !!window.cloudinary?.videoPlayer;
+
+	export let videoElement: HTMLVideoElement | undefined = undefined;
+	export let player: CloudinaryVideoPlayer | undefined = undefined;
+
+	$: if (videoElement && loaded && !player) {
+		player = window.cloudinary?.videoPlayer?.(videoElement, options);
+
+		player?.on('error', () =>
+			videoPlayerOptions.onError?.({
+				player: player!,
+				video: videoElement!,
+			}),
+		);
+
+		player?.on('loadeddata', () =>
+			videoPlayerOptions.onDataLoad?.({
+				player: player!,
+				video: videoElement!,
+			}),
+		);
+
+		player?.on('loadedmetadata', () =>
+			videoPlayerOptions.onMetadataLoad?.({
+				player: player!,
+				video: videoElement!,
+			}),
+		);
+
+		player?.on('pause', () =>
+			videoPlayerOptions.onPause?.({
+				player: player!,
+				video: videoElement!,
+			}),
+		);
+
+		player?.on('play', () =>
+			videoPlayerOptions.onPlay?.({
+				player: player!,
+				video: videoElement!,
+			}),
+		);
+
+		player?.on('ended', () =>
+			videoPlayerOptions.onEnded?.({
+				player: player!,
+				video: videoElement!,
+			}),
+		);
 	}
 
 	onMount(() => {
-		if (!window.cloudinary?.videoPlayer) {
-			return loadCloudinary({ type: 'video', onLoad: handleOnLoad, onError: onLoadError });
-		}
-		return handleOnLoad();
+		loadScript({
```

Comment:
something im doing in the Next.js version and will do in the Astro version is when i detect the client is idle, i create a new instance of the upload widget. i do this 1) to avoid loading all the extra JS in during page load time and 2) so that i can still load that JS in but after the initial load, with the goal of helping perf. doesnt look like this was doing that before 

https://github.com/cloudinary-community/next-cloudinary/blob/main/next-cloudinary/src/components/CldUploadWidget/CldUploadWidget.tsx#L128

feel free to test if this _actually_ helps or not but i suspect it would

Reply:
it could be worth doing this, I'd want to confirm that it helps first - and I think it makes sense to do that in a minor release

https://github.com/cloudinary-community/svelte-cloudinary/issues/129